src/v/storage/segment_utils.cc

```diff
@@ -552,7 +552,9 @@ ss::future<ss::lw_shared_ptr<segment>> make_concatenated_segment(
               "Aborting compaction of closed segment: {}", *segment));
         }
     }
-    co_await write_concatenated_compacted_index(path, segments, cfg);
+    auto compacted_idx_path = compacted_index_path(path);
+    co_await write_concatenated_compacted_index(
+      compacted_idx_path, segments, cfg);
     // concatenation process
```

Comment:
could you clarify your commit message?

> Concatenated compacted index has to have `.compacted_index` extension.
Previously concatenated compacted index was stored in incorrect path
which prevented it from being used when executing adjacent segments
compaction.

it looks like in the previous version of this code the index was written to the same file name as the segment data file and then was immediately overwritten by the segment data file. but given that, it's not clear that the requirement of ending in ".compacted_index" is required since the staging index files are renamed with proper extension in `internal::transfer_segment`.

I think that updating the commit message and renaming the staging index file to `.compacted_index.staging` is all that's needed. am I missing something?